main.go

```diff
@@ -42,6 +45,25 @@ func init() {
 	// +kubebuilder:scaffold:scheme
 }
 
+func sanitizeClusterDomainInput(clusterDomain string) string {
+	if len(clusterDomain) == 0 {
+		return ""
+	}
+
+	match, _ := regexp.MatchString("^\\.?[a-zA-Z-]+(\\.[a-zA-Z-]+)*$", clusterDomain) // Allow-list expression
```

Comment:
This regex is not general enough, for example, it doesn't allow numbers within domain names.

Reply:
Good catch! I'll build an expression to match following the [RFC 1035](https://datatracker.ietf.org/doc/html/rfc1035#section-2.3.1), except the 63 character limitation per label (hostname, subdomain, etc.) since it'll make a bit awkard to read to the regex, and it should be validated by Kubernetes itself, during the cluster initialisation process.